### PR TITLE
#45 add reading of multiple cookies

### PR DESCRIPTION
**Pull request by krzyk**
#45 
* `HtHeaders` now is a map of key-> list of values
* `HtCookies` can now have multiple cookies

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#51](https://github.com/yegor256/cactoos-http/issues/51) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C63314D6Z/1361059170); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

File: src/main/java/org/cactoos/http/HtHeaders.java [hunk 1]
```diff
@@ -38,25 +41,27 @@
...
-            final Map<String, String> map = new HashMap<>(lines.length - 1);
+            final Map<String, List<String>> map =
+                new HashMap<>(lines.length - 1);
             for (int idx = 1; idx < lines.length; ++idx) {
                 final String[] parts = lines[idx].split(":", 2);
-                map.put(
+                map.merge(
                     parts[0].trim().toLowerCase(Locale.ENGLISH),
-                    parts[1].trim()
+                    new ListOf<>(parts[1].trim()),
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/http/HtHeaders.java]: @krzyk Same of above here, use cactoos `TrimmedText` to trim this string.

File: src/main/java/org/cactoos/http/HtHeaders.java [hunk 2]
```diff
@@ -38,25 +41,27 @@
...
      * @param head Response head part
      */
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public HtHeaders(final Input head) {
         super(() -> {
             final String[] lines = new TextOf(head).asString().split("\r\n");
-            final Map<String, String> map = new HashMap<>(lines.length - 1);
+            final Map<String, List<String>> map =
+                new HashMap<>(lines.length - 1);
             for (int idx = 1; idx < lines.length; ++idx) {
```
Comment (paulodamaso) [inline src/main/java/org/cactoos/http/HtHeaders.java]: @krzyk I think that here you could use cactoos `SplitText` and its `iterator` method.

**Comment by krzyk** [comment]
@paulodamaso updated, pls take a look

**Comment by paulodamaso** [comment]
@rultor merge please

File: src/main/java/org/cactoos/http/HtContentType.java [hunk 3]
```diff
@@ -40,7 +42,7 @@
  * @version $Id$
  * @since 0.1
  */
-public final class HtContentType implements Scalar<String> {
+public final class HtContentType implements Scalar<List<String>> {
```
Comment (llorllale) [inline src/main/java/org/cactoos/http/HtContentType.java]: @krzyk just want to note that this is kind of weird, but I guess that nothing stops a sender to insert multiple `Content-Type` headers.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#51](https://github.com/yegor256/cactoos-http/issues/51) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request merged**